File: guidgen.go
package main

import (
    "fmt"
    "crypto/rand"
)

func main() {
    var numGUIDs int

    // Prompt the user for the number of GUIDs
    fmt.Println("Welcome to GUID Gen by Xanny.")
    fmt.Println("How many GUIDs do you need today? ")
    for {
        _, err := fmt.Scanln(&numGUIDs)
        if err != nil || numGUIDs <= 0 {
            fmt.Println("Invalid input. Please enter a number above zero.")
            continue
        }
        break
    }

    // Generate and print GUIDs
    fmt.Println("Generated GUIDs:")
    for i := 0; i < numGUIDs; i++ {
        guid, err := generateGUID()
        if err != nil {
            fmt.Println("Error generating GUID:", err)
            continue
        }
        fmt.Println(guid)
    }
}

// Generate a random GUID (version 4)
func generateGUID() (string, error) {
    b := make([]byte, 16)
    _, err := rand.Read(b)
    if err != nil {
        return "", err
    }

    // Set version bits (4th most significant bit to 1)
    b[4] = (b[4] | 0x80) & 0xaf

    // Set variant bits (most significant 2 bits to 10)
    b[8] = (b[8] | 0x80) & 0xbf

    return fmt.Sprintf("%x-%x-%x-%x-%x", b[0:4], b[4:6], b[6:8], b[8:10], b[10:]), nil
}

File: guidgen.py
import uuid

def generate_guids(num_guids):
  """
  Generates and returns a list of the specified number of random UUIDs (version 4).
  """
  if num_guids <= 0:
    print("Please enter a positive number of GUIDs to generate.")
    return

  guids = []
  for _ in range(num_guids):
    guids.append(str(uuid.uuid4()))

  return guids

# Prompt the user for the number of GUIDs
while True:
  try:
    num_guids = int(input("Welcome to GUID Gen by Xanny. \nHow many GUIDs do you need today? "))
    break
  except ValueError:
    print("Invalid input. Please enter a number above zero.")

# Generate the GUIDs and print them
guids = generate_guids(num_guids)
if guids:
  print("\nGenerated GUIDs: \n")
  for guid in guids:
    print(guid)
